make/verify-dependencies.mk

```diff
@@ -0,0 +1,16 @@
+.PHONY: verify-dependencies
+## Runs commands to verify after the updated dependecies of toolchain-common/API(go mod replace), if the repo needs any changes to be made
+verify-dependencies: tidy vet compile test lint-go-code
+
+.PHONY: tidy
+tidy: 
+	go mod tidy
+
+.PHONY: vet
+vet:
+	go vet ./...
+
+.PHONY: compile
+compile:
+	go test ./... -skip '.*'
```

Comment:
I already left a comment in the member-operator - everything is related to `go` commands and it's also verifying validity of the go dependencies, let's move it to `go.mk` file

Reply:
i thought about keeping the dependency check separate but adding everything related to go in one file makes sense.. moved it

Reply:
in go.mk the commands are related to build, also registration service doesn't have go.mk it has build.mk for the same.. i was thinking to either have a separate file which is same across all repos, or rename build.mk file in registration service to go.mk (same as other repos) WDYT?

Reply:
let's move it to `go.mk` - that's the place where I would look for any commands/targets related to go binary - and as you mentioned, rename `build.mk` to `go.mk` in reg-service to keep it consistent :+1: